addons/payment_mercado_pago/models/payment_acquirer.py

```diff
@@ -0,0 +1,125 @@
+        })
+
+    # === BUSINESS METHODS === #
+
+    def _mercado_pago_make_request(self, endpoint, payload=None, method='POST'):
+        """ Make a request to Mercado Pago API.
+
+        :param str endpoint: The endpoint to request.
+        :param dict payload: The payload to send.
+        :param str method: The HTTP method to use.
+
+        :return: The response of the request.
+        :rtype: dict.
+        """
+        self.ensure_one()
+
+        headers = {
+            'Authorization': f'Bearer {self.mercado_pago_access_token}',
+            'Content-Type': 'application/json'
+        }
+
+        request_url = urls.url_join(MP_BASE_URL, endpoint)
+
+        _logger.info(
+            "sending request to %s:\n%s\n%s",
+            request_url, pprint.pformat(headers), pprint.pformat(payload or {})
+        )
+
+        try:
+            response = requests.request(
+                method, request_url, data=json.dumps(payload), headers=headers, timeout=60
+            )
+
+            if not response.ok and 400 <= response.status_code < 500:
+                # Mercado Pago errors have status in the 400s values
+                try:
+                    response.raise_for_status()
+                except requests.exceptions.HTTPError:
+                    _logger.exception("invalid API request at %s with data %s",
+                                      request_url, pprint.pformat(payload))
+
+                    response = json.loads(response.content)
+
+                    error_type = response.get('error', '')
+                    error_msg = response.get('message', '')
+
+                    raise ValidationError(
+                        "Mercado Pago: " + _(
+                            "The communication with the API failed.\nMercado Pago gave us the "
+                            "following info about the problem:\n'%s': '%s'",
+                            error_type, error_msg
+                        )
+                    )
+
+            return json.loads(response.content)
+        except requests.exceptions.ConnectionError:
+            _logger.exception("unable to reach endpoint at %s", request_url)
+            raise ValidationError(
+                "Mercado Pago: " + _("Could not establish the connection to the API.")
+            )
+
+    @api.model
+    def _get_compatible_acquirers(self, *args, currency_id=None, is_validation=False, **kwargs):
+        """ Override of payment to unlist MP acquirers for unsupported currencies. """
+
+        acquirers = super()._get_compatible_acquirers(
+            *args, currency_id=currency_id, is_validation=is_validation, **kwargs
+        )
+
+        currency = self.env['res.currency'].browse(currency_id).exists()
+        if (currency and currency.name not in ALLOWED_CURRENCIES) or is_validation:
+            acquirers = acquirers.filtered(
+                lambda a: a.provider != 'mercado_pago'
+            )
+
+        return acquirers
```

Comment:
Method should be before `_mercado_pago_make_request`, it's used earlier in the payment flow (and quite critical when wanting to understand why the acquirer isn't shown :D)